testnet/stacks-node/src/tests/nakamoto_integrations.rs

```diff
@@ -5185,51 +5185,21 @@ fn clarity_burn_state() {
                 vec![&Value::UInt(burn_block_height)],
             );
             result.expect_result_ok().expect("Read-only call failed");
-
-            // Submit a tx for the next block (the next block will be a new tenure, so the burn block height will increment)
-            let call_tx = tests::make_contract_call(
-                &sender_sk,
-                sender_nonce,
-                tx_fee,
-                &sender_addr,
-                contract_name,
-                "bar",
-                &[Value::UInt(burn_block_height + 1)],
-            );
-            sender_nonce += 1;
-            submit_tx(&http_origin, &call_tx);
         }
 
         let commits_before = commits_submitted.load(Ordering::SeqCst);
-        next_block_and_process_new_stacks_block(&mut btc_regtest_controller, 60, &coord_channel)
-            .unwrap();
+        next_block_and_mine_commit(
+            &mut btc_regtest_controller,
+            60,
+            &coord_channel,
+            &commits_submitted,
+        )
+        .unwrap();
 
         let info = get_chain_info(&naka_conf);
         burn_block_height = info.burn_block_height as u128;
         info!("Expecting burn block height to be {}", burn_block_height);
 
-        // Assert that the contract call was successful
-        test_observer::get_mined_nakamoto_blocks()
-            .last()
-            .unwrap()
-            .tx_events
-            .iter()
-            .for_each(|event| match event {
-                TransactionEvent::Success(TransactionSuccessEvent { result, fee, .. }) => {
-                    // Ignore coinbase and tenure transactions
-                    if *fee == 0 {
-                        return;
-                    }
-
-                    info!("Contract call result: {}", result);
-                    result.clone().expect_result_ok().expect("Ok result");
-                }
-                _ => {
-                    info!("Unsuccessful event: {:?}", event);
-                    panic!("Expected a successful transaction");
-                }
-            });
-
```

Comment:
Is this just deleting the thing the test is supposed to be testing?

Reply:
Yes because it is impossible to test it. They are assuming that the stacks block will be processed AFTER the burn block. Whihc is impossible to ensure. And it already tests interim blocks after which test the same thing which it was TRYING to assert which is that the stacks blocks mined AFTER the burn block are as expected.

Reply:
Ah ok, I see. But the thing that we are not testing still then is when the transaction is in the tenure change block. That's important to test because it was buggy in the early implementation. Maybe we can use the fault injection locks that we've put in to force that scenario?

Reply:
I don't understand what you mea. it still does the get_mined_nakamoto_blocks for all mined nakamoto blocks. which will test the same thing as it did before. just won't test it prematurely.

Reply:
I don't think it ever tested the tenure change block...?

Reply:
I think we're guaranteeing that the call to `bar` will never be in the first block of a tenure with this change.

Reply:
That's what this deleted code was trying to do, but it would be flaky.

Reply:
Ah you want next_block_and_commit rather than new_stacks_block being called as well. Can make that change pretty easily. Not sure it guarantees what you are asking though.

Reply:
To ensure we can test that, I think we need to:
1. Pause block production
2. Submit the call to `bar`
3. Mine a bitcoin block
4. Resume block production

Reply:
Maybe you could do it with a long `min_time_between_blocks_ms`?

Reply:
EDIT: OHHHH next_block_and_mine_commit will force a stacks block to be mined. Okay one sec.